src/app/model/ChessEngine.ts

```diff
@@ -155,14 +168,14 @@ export class ChessEngine implements ChessModel {
     return this.isKingChecked(simulatedChessboard, this.currentSide);
   }
 
-  public pieceMovesNotCausingAllyKingCheckmate(position: Square): Square[] {
+  public pieceMovesNotCausingAllyKingCheck(position: Square): Square[] {
     const initialPossibleMoves = this.board.onPositionPiece(position)?.possibleMoves(position, this.board) ?? [];
     const filteringFunction = (onePossibleMove: Square) => !this.willBeKingChecked(position, onePossibleMove);
     return initialPossibleMoves.filter(filteringFunction);
   }
 
   public possibleMoves(position: Square): Square[] {
-    return this.pieceMovesNotCausingAllyKingCheckmate(position);
+    return this.pieceMovesNotCausingAllyKingCheck(position);
   }
```

Comment:
tego trochę nie rozumiem, w sensie chyba powinno być tak, że jakaś bierka ma swoje possible moves zależne od pozycji i teraz zadaniem chess enigine będzie zrewidować te ruchy pod kątem możliwości 'wystawienia' swojego króla, czyli inaczej mówiąc chessengine powinien skrócić listę possible moves o te które są niebezpieczne dla króla

Reply:
Tak, i dokładnie tak się dzieje tuta :) 
Tylko powinniśmy przetestować, że possibleMoves dokładnie to robi. `pieceMovesNotCausingAllyKingCheck` jest nadmiarowa, a z pewnością nie powinna być ani testowana ani publiczna.

Reply:
done